embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/EmbraceDeliveryCacheManager.kt

```diff
@@ -62,9 +62,14 @@ internal class EmbraceDeliveryCacheManager(
             val sessionId = sessionMessage.session.sessionId
             val writeSync = snapshotType == SessionSnapshotType.JVM_CRASH
             val snapshot = snapshotType == SessionSnapshotType.PERIODIC_CACHE
-            saveBytes(sessionId, writeSync, snapshot) { filename: String ->
+
+            saveSessionBytes(sessionId, writeSync, snapshot) { filename: String ->
                 Systrace.traceSynchronous("serialize-session") {
-                    cacheService.writeSession(filename, sessionMessage)
+                    if (!cachedSessions.containsKey(sessionId)) {
```

Comment:
Access to `cachedSessions` doesn't seem safe if this can get called via multiple threads. This seems like a more general issue throughout the class.

Reply:
Yup. We require the caller to be synchronized. But I was able to eliminate this check in the new PR, so this specific issue won't be a problem. That said, I'll take a look at the access throughout this class to see if anything needs to be changed.